cache/sizedlru/sizedlru.go

```diff
@@ -0,0 +1,111 @@
+// Package sizedlru implements an LRU cache that bounds the total space occupied
+// by the items it contains.
+
+package sizedlru
+
+import (
+	"container/list"
+)
+
+type SizedItem interface {
+	Size() int64
+}
+
+// SizedLRU is an LRU cache that will keep its total size below maxSize by evicting
+// items. If maxSize is zero, eviction is disabled.
```

Comment:
I am thinking we might not even want to support this. IMHO there's no use case for an unbounded cache and one can always set maxSize to a really big value (or disk size). Wdyt?

Reply:
removed